pkg/build/build.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	dockerlib "github.com/fsouza/go-dockerclient"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func BuildDockerImage(dockerfile, image, context string) error {
```

Comment:
Can you please add comments to this function?
What is the difference between this and `func (c *Build) BuildImage(dockerfile, image, context string) error` ?